Plugins/Public/help_expanded/Main.cpp

```diff
@@ -0,0 +1,302 @@
+// Expanded Help Menu - Customisable Help Commands
+// By Laz, w/ help from @Alex.
+//
+// This is free software; you can redistribute it and/or modify it as
+// you wish without restriction. If you do then I would appreciate
+// being notified and/or mentioned somewhere.
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Includes
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include <windows.h>
+#include <stdio.h>
+#include <string>
+#include <time.h>
+#include <math.h>
+#include <list>
+#include <map>
+#include <algorithm>
+#include <FLHook.h>
+#include <plugin.h>
+#include <PluginUtilities.h>
+#include "Main.h"
+
+#include "../hookext_plugin/hookext_exports.h"
+
+/// A return code to indicate to FLHook if we want the hook processing to continue.
+PLUGIN_RETURNCODE returncode;
+
+void LoadSettings();
+
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+	srand((uint)time(0));
+	// If we're being loaded from the command line while FLHook is running then
+	// set_scCfgFile will not be empty so load the settings as FLHook only
+	// calls load settings on FLHook startup and .rehash.
+	if(fdwReason == DLL_PROCESS_ATTACH)
+	{
+		if (set_scCfgFile.length()>0)
+			LoadSettings();
+	}
+	else if (fdwReason == DLL_PROCESS_DETACH)
+	{
+	}
+	return true;
+}
+
+/// Hook will call this function after calling a plugin function to see if we the
+/// processing to continue
+EXPORT PLUGIN_RETURNCODE Get_PluginReturnCode()
+{
+	return returncode;
+}
+
+struct HELPSTRUCT
+{
+	wstring wscTitle; // What will the title of our text box be called?
+	wstring wscContent; // What content will it hold?
+};
+
+bool bPluginEnabled = true; // So we can disable it with ease.
+map<wstring, HELPSTRUCT> mapHelp; 
+list<wstring> lstHelpCommands; // So we have a list of help commands that can be iterated over
```

Comment:
we actually don't need this, it's only used once and what we can do there is loop over mapHelp, looking at ->first